src/Dialect/ONNX/ONNXOps.td

```diff
@@ -112,6 +112,17 @@ def ONNXMaxPoolSingleOutOp: ONNX_Op<"MaxPoolSingleOut",
            DefaultValuedAttr<I64Attr, "0">:$storage_order,
            OptionalAttr<I64ArrayAttr>:$strides);
   let results = (outs AnyTypeOf<[AnyMemRef, AnyTensor]>:$o_Y);
+  let extraClassDeclaration = [{
+    static int getNumberOfOperands() {
+      return 1;
+    }
+    static int getNumberOfResults() {
+      return 1;
+    }
+    static std::vector<int> getTypeMap() {
+      return {0};
+    }
+  }];
```

Comment:
This looks like it's copied several times, with just a few numbers changing, could this be factored out into a template somehow? It's fine if not but it would be nice to have it that way if possible.

Reply:
These attributes vary with operations. But  these values are most common ones. This is for Tablegen. I feel it is better to express them explicitly. I will look into trait inherit to specify the common value as a default. You can see that it is not logically default, but just common value.